### PR TITLE
[SPARK-25755][SQL][Test] Supplementation of non-CodeGen unit tested for BroadcastHashJoinExec

### PR DESCRIPTION
**Pull request by heary-cao**
## What changes were proposed in this pull request?

Currently, the BroadcastHashJoinExec physical plan supports CodeGen and non-codegen, but only CodeGen code is tested in the unit tests of InnerJoinSuite、OuterJoinSuite、ExistenceJoinSuite, and non-codegen code is not tested. This PR supplements this part of the test.
 
## How was this patch tested?

add new unit tested.

### CODE REVIEW

**Comment by heary-cao** [comment]
cc @gatorsmile, @HyukjinKwon

**Comment by kiszk** [comment]
cc @maropu

**Comment by maropu** [comment]
ok to test

File: sql/core/src/test/scala/org/apache/spark/sql/execution/joins/ExistenceJoinSuite.scala [hunk 1]
```diff
@@ -122,19 +122,22 @@ class ExistenceJoinSuite extends SparkPlanTest with SharedSQLContext {
...
-                leftKeys, rightKeys, joinType, BuildRight, boundCondition, left, right)),
-            expectedAnswer,
-            sortAnswers = true)
-          checkAnswer2(leftRows, rightRows, (left: SparkPlan, right: SparkPlan) =>
-            EnsureRequirements(left.sqlContext.sessionState.conf).apply(
-              createLeftSemiPlusJoin(BroadcastHashJoinExec(
-                leftKeys, rightKeys, leftSemiPlus, BuildRight, boundCondition, left, right))),
-            expectedAnswer,
-            sortAnswers = true)
+        Seq("false", "true").foreach { v =>
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/joins/ExistenceJoinSuite.scala]: nit: v -> codegenEnabled

**Comment by heary-cao** [comment]
retest this please

**Comment by heary-cao** [comment]
cc @maropu, @kiszk

**Comment by cloud-fan** [comment]
+1

**Comment by kiszk** [comment]
Is it better to apply this util method to others (e.g. `DataFrameRangeSuite.scala` and `DataFrameAggregateSuite.scala`)?

File: sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala [hunk 2]
```diff
@@ -166,6 +167,17 @@ private[sql] trait SQLTestUtilsBase
     super.withSQLConf(pairs: _*)(f)
   }
 
+  /**
+   * A helper function for turning off/on codegen.
+   */
+  protected def withCodegenTurnOffAndOn(f: String => Unit): Unit = {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala]: nit: `withWholeStageCodegenOnAndOff`

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 3]
```diff
@@ -459,6 +459,7 @@ private[spark] class Executor(
           threadMXBean.getCurrentThreadCpuTime
         } else 0L
 
+
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala]: unnecessary change

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 4]
```diff
@@ -669,22 +669,20 @@ class DataFrameAggregateSuite extends QueryTest with SharedSQLContext {
...
-        checkAnswer(spark.emptyDataFrame.groupBy().agg(emptyAgg), Seq(Row()))
-        checkAnswer(spark.emptyDataFrame.groupBy().agg(count("*")), Seq(Row(0)))
-        checkAnswer(spark.emptyDataFrame.dropDuplicates().agg(emptyAgg), Seq(Row()))
-        checkAnswer(spark.emptyDataFrame.dropDuplicates().groupBy().agg(emptyAgg), Seq(Row()))
-        checkAnswer(spark.emptyDataFrame.dropDuplicates().groupBy().agg(count("*")), Seq(Row(0)))
-
-        // global aggregation is converted to grouping aggregation:
-        assert(spark.emptyDataFrame.dropDuplicates().count() == 0)
-      }
+      s"results when codegen is enabled: $codegenEnabled") {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala]: we can add the test name postfix in `withWholeStageCodegenOnAndOff`, so that the caller side only need to provide a base name. e.g.
```
def withWholeStageCodegenOnAndOff(testName: String)(f: String => Unit): Unit = {
  Seq("false", "true").foreach { enabled =>
    test(s"$testName (whole-stage-codegen ${if enabled "on" else "off"})") ...
  }
}
```

**Comment by cloud-fan** [comment]
LGTM

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by heary-cao** [comment]
thanks @cloud-fan

File: sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala [hunk 5]
```diff
@@ -65,6 +66,20 @@ private[sql] trait SQLTestUtils extends SparkFunSuite with SQLTestUtilsBase with
     }
   }
 
+  /**
+   * A helper function for turning off/on codegen.
+   */
+  protected def testWithWholeStageCodegenOnAndOff(testName: String)(f: String => Unit): Unit = {
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala]: not a big deal at all but just wanted to say. Looks we could just `f: () => Unit`. `codegenEnabled` as an argument is used at two places.

One place is just for clue and it can be removed since test name already describes if whole stage codegen is off or not.

The other place looks it needs but I think we can get read it from sql conf.
